modules/drivers/bigquery/src/metabase/driver/bigquery.clj

```diff
@@ -200,35 +225,30 @@
                 (dissoc :database-type :database-position)))]
       (respond
        {:cols columns}
-       (for [^TableRow row (.getRows response)]
-         (for [[^TableCell cell, parser] (partition 2 (interleave (.getF row) parsers))]
-           (when-let [v (.getV cell)]
+       (letfn [(fetch-page [^GetQueryResultsResponse response]
+                           (lazy-cat
+                            (.getRows response)
+                            (when-let [next-page-token (.getPageToken response)]
+                              (with-finished-response [next-resp (get-query-results (database->client database)
+                                                                                    (.getProjectId (.getJobReference response))
+                                                                                    (.getJobId (.getJobReference response))
+                                                                                    next-page-token)]
+                                (fetch-page next-resp)))))]
+        (for [^TableRow row (fetch-page response)]
+          (for [[^TableCell cell, parser] (partition 2 (interleave (.getF row) parsers))]
```

Comment:
Wouldn't `(map (fn [cell parser] ...) (.getF row) parsers)` be clearer here?

Reply:
Maybe? The nested `for` loops were already present before my change - I added the `fetch-page` stuff.

Reply:
I did the nested `for` because I thought *that* was clearer but I can see how `map` would be clearer... I could go either way if you want to change it